src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/FunctionCallUtilities.cs

```diff
@@ -136,12 +165,19 @@ public static JsonElement FormatFunctionResultAsJsonElement(object? result, Json
     /// <param name="parameterMetadata">The parameter metadata from which to infer the schema.</param>
     /// <param name="functionMetadata">The containing function metadata.</param>
     /// <param name="options">The global <see cref="JsonSerializerOptions"/> governing serialization.</param>
+    /// <param name="includeTypeInEnumSchemas">Whether to include the type keyword in schemas for enum types.</param>
+    /// <param name="disallowAdditionalProperties">Whether to emit an additionalProperties keyword set to false.</param>
     /// <returns>A JSON schema document encoded as a <see cref="JsonElement"/>.</returns>
     public static JsonElement InferParameterJsonSchema(
         AIFunctionParameterMetadata parameterMetadata,
         AIFunctionMetadata functionMetadata,
-        JsonSerializerOptions? options)
+        JsonSerializerOptions? options,
+        bool includeTypeInEnumSchemas = false,
+        bool disallowAdditionalProperties = false)
```

Comment:
What's the liklihood we'll want to expose more configuration here? Wondering if this should instead be an options bag, or a flags enum.

Reply:
It's something I've been thinking about. It's likely we'll need more of these in the future as we encounter more vendors imposing their own restrictions on accepted schemas.

Reply:
Moved to a separate `JsonSchemaInferenceOptions` class.